app/code/Magento/Cms/Test/Integration/Model/Wysiwyg/Images/GetInsertImageContentTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Cms\Test\Integration\Model\Wysiwyg\Images;
+
+use Magento\Cms\Helper\Wysiwyg\Images as ImagesHelper;
+use Magento\Cms\Model\Wysiwyg\Images\GetInsertImageContent;
+use Magento\TestFramework\Helper\Bootstrap;
+use PHPUnit\Framework\TestCase;
+
+class GetInsertImageContentTest extends TestCase
+{
+    private const TEST_IMAGE_FILE = '/test-image.jpg';
+
+    /**
+     * @var GetInsertImageContent
+     */
+    private $getInsertImageContent;
+
+    /**
+     * @var ImagesHelper
+     */
+    private $imagesHelper;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp(): void
+    {
+        $this->getInsertImageContent = Bootstrap::getObjectManager()->get(GetInsertImageContent::class);
+        $this->imagesHelper = Bootstrap::getObjectManager()->get(ImagesHelper::class);
+    }
+
+    /**
+     * Test for GetInsertImageContent::execute
+     *
+     * @dataProvider imageDataProvider
+     * @param string $encodedFilename
+     * @param bool $forceStaticPath
+     * @param bool $renderAsTag
+     * @param int|null $storeId
+     */
+    public function testExecute(
+        string $encodedFilename,
+        bool $forceStaticPath,
+        bool $renderAsTag,
+        ?int $storeId = null
+    ): void {
+        $getImageForInsert = $this->getInsertImageContent->execute(
+            $encodedFilename,
+            $forceStaticPath,
+            $renderAsTag,
+            $storeId
+        );
+
+        if (!$forceStaticPath) {
```

Comment:
The goal of the test is to verify the correct return from the model based on the input parameters.
Instead of keeping complex logic in the test, I'd pass the expected result from the data provider. That should simplify the test to one-line `$this->assertEquals($expectedResult, this->getInsertImageContent->execute(<params>)`.

Having the expected result in the data provider will also add a great benefit of clear understanding on how the model is actually working.

In case the result contains the dynamic parts depending on configuration, like hostname, these can be replaced in the expected result string before comparison.